src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -12,13 +12,22 @@ import {BreadcrumbsSeparator} from './BreadcrumbsSeparator';
 
 import './Breadcrumbs.scss';
 
-export interface BreadcrumbsItem {
+type BaseBreadcrumbsItem = {
     text: string;
     action: (event: React.MouseEvent<HTMLElement, MouseEvent> | KeyboardEvent) => void;
-    href?: string;
     items?: BreadcrumbsItem[];
     title?: string;
-}
+};
+
+type LinkBreadcrumbsItem = {
+    href: string;
+} & BaseBreadcrumbsItem;
+
+type ButtonBreadcrumbsItem = {
+    href?: undefined;
+} & BaseBreadcrumbsItem;
+
+export type BreadcrumbsItem = LinkBreadcrumbsItem | ButtonBreadcrumbsItem;
```

Comment:
Isn't the same as before?

Reply:
It was types product, became types sum. Its more powerful way for union declaration.

Reply:
> I mean there is no difference, `href` is optional anyway. What's the point of this change?

Oh, I figure out, why it wasn't obvious to use this kind of approach. Lost it while rebasing, it should have been:
```
type BaseBreadcrumbsItem = {
    text: string;
    items?: BreadcrumbsItem[];
    title?: string;
};

type LinkBreadcrumbsItem = {
    href: string;
    action?: (event: React.MouseEvent<HTMLElement, MouseEvent> | KeyboardEvent) => void;
} & BaseBreadcrumbsItem;

type ButtonBreadcrumbsItem = {
    href?: undefined;
    action: (event: React.MouseEvent<HTMLElement, MouseEvent> | KeyboardEvent) => void;
} & BaseBreadcrumbsItem;
```
So, we describe 2 categories of breadcrumb items: the one is like regular anchor(and should have the href prop), the two is like button, that should have some action(action prop).
So, if user want to create Breadcrumb, that is not like regular link, he will be forced(by typescript) to specify the action prop.
And if the developer(of uikit) needs to add a new behavior for only one of the types of breadcrumb item(e.g. the target attribute for anchor), then he modifies only LinkBreadcrumbsItem and any time in the code he will know, that he works with certain type of item(due to the href prop check).